src/view/com/lightbox/ImageViewing/index.tsx

```diff
@@ -169,13 +173,190 @@ const styles = StyleSheet.create({
+      return {}
+    }
+    return {
+      transform: [
+        {
+          scale: interpolate(
+            openProgress.value,
+            [0, 1],
+            [initialTransform.scale, 1],
+          ),
+        },
+        {
+          translateX: interpolate(
+            openProgress.value,
+            [0, 1],
+            [initialTransform.translateX, 0],
+          ),
+        },
+        {
+          translateY: interpolate(
+            openProgress.value,
+            [0, 1],
+            [initialTransform.translateY, 0],
+          ),
+        },
+      ],
+    }
+  })
+  const insets = useSafeAreaInsets()
+  const t = useTheme()
+  const showSplash =
+    !PlatformInfo.getIsReducedMotionEnabled() && initialTransform && !isReady
+
+  if (!props.visible) {
+    return null
+  }
+
+  return (
+    <View
+      aria-modal
+      accessibilityViewIsModal
+      style={{
+        position: 'absolute',
+        top: isAndroid ? insets.top : 0,
+        left: 0,
+        bottom: 0,
+        right: 0,
+      }}>
+      {showSplash && (
+        <Animated.View
+          style={[
+            {
+              position: 'absolute',
+              top: 0,
+              left: 0,
+              bottom: 0,
+              right: 0,
+              zIndex: 1,
+              pointerEvents: 'none',
+            },
+          ]}>
+          {props.thumbDims && isThumbLoaded && (
+            <View
+              style={{
+                position: 'absolute',
+                top: props.thumbDims.pageY - (isAndroid ? insets.top : 0),
+                left: props.thumbDims.pageX,
+                width: props.thumbDims.width,
+                height: props.thumbDims.height,
+                backgroundColor: t.atoms.bg.backgroundColor,
+              }}
+            />
+          )}
+          <Animated.View
+            style={[
+              {
+                position: 'absolute',
+                top: 0,
+                left: 0,
+                bottom: 0,
+                right: 0,
+                backgroundColor: 'black',
+              },
+              backgroundStyle,
+            ]}
+          />
+          <AnimatedImage
+            contentFit="contain"
+            source={{uri: initialImage.thumbUri}}
+            onLoad={() => setIsThumbLoaded(true)}
+            style={[
+              {
+                position: 'absolute',
+                top: 0,
+                left: 0,
+                bottom: 0,
+                right: 0,
+              },
+              animatedStyle,
+            ]}
+          />
+        </Animated.View>
+      )}
+      <Animated.View
+        style={{
+          position: 'absolute',
+          top: 0,
+          left: 0,
+          right: 0,
+          bottom: 0,
+          pointerEvents: showSplash ? 'none' : 'auto',
+          opacity: showSplash ? 0 : 1,
+        }}>
+        <ImageViewing
+          key={props.initialImageIndex}
+          {...props}
+          onLoad={() => {
```

Comment:
This might be too conservative since it'll just get stuck if the network is slow. Maybe worth adding a safeguard timeout. Or it would require a bigger rework to integrate with the slider and dismiss gesture inside.